OpenRA.Game/Traits/BodyOrientation.cs

```diff
@@ -62,7 +63,7 @@ public BodyOrientation(ActorInitializer init, BodyOrientationInfo info)
 				if (info.QuantizedFacings >= 0)
 					return info.QuantizedFacings;
 
-				var qboi = self.Info.Traits.GetOrDefault<IQuantizeBodyOrientationInfo>();
+				var qboi = self.Info.Traits.WithInterface<IQuantizeBodyOrientationInfo>().FirstOrDefault();
```

Comment:
This is bogus for two reasons:

1.  You're unconditionally taking the first implementation of this trait, regardless of whether it is active or not.
2.  Fixing 1. doesn't help, because a bunch of things implicitly depend on `IQuantizeBodyOrientationInfo` not changing.

A better idea might be to move `IQuantizeBodyOrientationInfo` onto `RenderSprites` / `RenderVoxels`, but i'll leave it to you to work out how to make `RenderSprites` work out the number of facings it should use.